test/test_grid_addboundaryset.jl

```diff
@@ -0,0 +1,39 @@
+@testset "grid addboundaryset" begin
+#                           (11)
+#                   (10)+-----+-----+(12)
+#                       |  5  |  6  |
+#                   (7) +-----+-----+(9)
+#                       |  3  |  4  |
+#                   (4) +-----+-----+(6)
+#                       |  1  |  2  |
+#                   (1) +-----+-----+(3)
+#                            (2)
+    grid = generate_grid(Quadrilateral,(2,3))
+    topology = ExclusiveTopology(grid)
+    #test boundary faces
+    addboundaryfaceset!(grid, topology, "b_bottom", x -> x[2] ≈ -1.0)
+    addboundaryfaceset!(grid, topology, "b_top", x -> x[2] ≈ 1.0)
+    addboundaryfaceset!(grid, topology, "b_right", x -> x[1] ≈ 1.0)
+    addboundaryfaceset!(grid, topology, "b_left", x -> x[1] ≈ -1.0)
+    addboundaryfaceset!(grid, topology, "nb_empty", x -> x[2] ≈ 0.0) #can use any rule that is false at all boundaries to test
+    @test grid.facesets["b_bottom"] == grid.facesets["bottom"]
+    @test grid.facesets["b_top"] == grid.facesets["top"]
+    @test grid.facesets["b_right"] == grid.facesets["right"]
+    @test grid.facesets["b_left"] == grid.facesets["left"]
+    @test grid.facesets["nb_empty"] == Set([])
+    #test boundary vertices
+    addboundaryvertexset!(grid, topology, "b_bottom", x -> x[2] ≈ -1.0)
+    addboundaryvertexset!(grid, topology, "b_top", x -> x[2] ≈ 1.0)
+    addboundaryvertexset!(grid, topology, "b_right", x -> x[1] ≈ 1.0)
+    addboundaryvertexset!(grid, topology, "b_left", x -> x[1] ≈ -1.0)
+    addboundaryvertexset!(grid, topology, "nb_empty", x -> x[2] ≈ 0.0) #can use any rule that is false at all boundaries to test
+    addvertexset!(grid, "bottom", x -> x[2] ≈ -1.0)
+    addvertexset!(grid, "top", x -> x[2] ≈ 1.0)
+    addvertexset!(grid, "right", x -> x[1] ≈ 1.0)
+    addvertexset!(grid, "left", x -> x[1] ≈ -1.0)
+    @test grid.vertexsets["b_bottom"] == grid.vertexsets["bottom"]
+    @test grid.vertexsets["b_top"] == grid.vertexsets["top"]
+    @test grid.vertexsets["b_right"] == grid.vertexsets["right"]
+    @test grid.vertexsets["b_left"] == grid.vertexsets["left"]
+    @test grid.vertexsets["nb_empty"] == Set([])
+end
```

Comment:
I think we have to work a bit more on the test. This one is too simple, because it does not contain any case which does not work for addfaceset/addvertexset . I think this test here would be better:
1. extract all boundary edges/faces/vertices individually via add*set! and merge the sets
2. check if this merged set matches addboundary*set!(grid, topology, "full_boundary", true)
This should be checked for triangular and tetrahedral meshes from the grid generator, because they contain the case that faces/edges can pass through the interior as in this example https://github.com/Ferrite-FEM/Ferrite.jl/discussions/565 .